src/StoreApi/Routes/CartUpdateCustomer.php

```diff
@@ -146,56 +145,4 @@ protected function get_customer_billing_address( \WC_Customer $customer ) {
 			'phone'      => $customer->get_billing_phone(),
 		];
 	}
-
-	/**
-	 * If there is a draft order, update the customer data within that also so the
-	 * cart and order are kept in sync.
-	 *
-	 * @param \WC_Customer     $customer Customer object.
-	 * @param \WP_REST_Request $request Request object.
-	 */
-	protected function maybe_update_order_from_customer( \WC_Customer $customer, \WP_REST_Request $request ) {
-		$draft_order = $this->get_draft_order();
-
-		if ( ! $draft_order ) {
-			return;
-		}
-
-		$draft_order->set_props(
-			[
-				'billing_first_name'  => $customer->get_billing_first_name(),
-				'billing_last_name'   => $customer->get_billing_last_name(),
-				'billing_company'     => $customer->get_billing_company(),
-				'billing_address_1'   => $customer->get_billing_address_1(),
-				'billing_address_2'   => $customer->get_billing_address_2(),
-				'billing_city'        => $customer->get_billing_city(),
-				'billing_state'       => $customer->get_billing_state(),
-				'billing_postcode'    => $customer->get_billing_postcode(),
-				'billing_country'     => $customer->get_billing_country(),
-				'billing_email'       => $customer->get_billing_email(),
-				'billing_phone'       => $customer->get_billing_phone(),
-				'shipping_first_name' => $customer->get_shipping_first_name(),
-				'shipping_last_name'  => $customer->get_shipping_last_name(),
-				'shipping_company'    => $customer->get_shipping_company(),
-				'shipping_address_1'  => $customer->get_shipping_address_1(),
-				'shipping_address_2'  => $customer->get_shipping_address_2(),
-				'shipping_city'       => $customer->get_shipping_city(),
-				'shipping_state'      => $customer->get_shipping_state(),
-				'shipping_postcode'   => $customer->get_shipping_postcode(),
-				'shipping_country'    => $customer->get_shipping_country(),
-				'shipping_phone'      => $customer->get_shipping_phone(),
-			]
-		);
-
-		/**
-		 * Fires when the Checkout Block/Store API updates an existing draft order from customer data.
-		 *
-		 * @param \WC_Order $draft_order Order object.
-		 * @param \WC_Customer $customer Customer object.
-		 * @param \WP_REST_Request $request Full details about the request.
-		 */
-		do_action( 'woocommerce_blocks_cart_update_order_from_customer_request', $draft_order, $customer, $request );
```

Comment:
@senadir this didn't make sense after moving all updates to cart...

Reply:
Since this was unreleased in https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/5345 we can avoid deprecation if we ship this in 6.6.0 instead.